src/components/grid/GridContainer.tsx

```diff
@@ -45,26 +46,27 @@ const GridContainer: FC<GridContainerProps> = (props: GridContainerProps) => {
   };
 
   const onLayoutChange = (newLayouts: Layout[]) => {
-    setLayouts(newLayouts);
+    setWidgetConfigs(getWidgetConfByLayout(newLayouts, widgetConfigs));
   };
+
   const isOnSettingPage = useLocation().pathname === RouterPaths.SETTINGS;
   /**
    * fire when a widget is moved and placed in a different position
    * @param newLayouts
    */
   const onDragStop = async (newLayouts: Layout[]) => {
-    setLayouts(newLayouts);
-    const isSuccess = await updateLayout(
+    const newWidgetConfs = getWidgetConfByLayout(newLayouts, widgetConfigs);
```

Comment:
As mentioned elsewhere also, cant you make widgetconfig service independent of layout? Getting the widgetconfig using a layout can be done in this class/file.